src/bandersnatch/mirror.py

```diff
@@ -913,6 +911,42 @@ async def download_file(
         return path
 
 
+_T = TypeVar("_T")
+
+
+async def _setup_diff_file(
+    storage_plugin: Storage, configured_path: str, append_epoch: bool
+) -> Path:
+    # save some typing. maybe a method to add to Storage?
+    async def storage_plugin_exec(fn: Callable[..., _T], *args: Any) -> _T:
+        return await storage_plugin.loop.run_in_executor(
+            storage_plugin.executor, fn, *args
+        )
```

Comment:
Yeah, I thought this function might work as a method on the base class? I didn’t look to see how common using the storage plugin this way is in the rest of the code. I supposed that most of the time one would call a method on the storage backend directly. But if it could be useful elsewhere the storage plugins could have this as a method, just wrapping `run_in_executor` for convenience.

Reply:
Finally did a quick Ctrl+F and I'm sold on adding an `async def run_in_executor(...)` helper method to the `Storage` plugin base class, but I do think it would be better in a later PR.